File: app/src/main/java/tw/y_studio/ptt/api/SetPostRankAPIHelper.java
package tw.y_studio.ptt.api;

import android.content.Context;

import okhttp3.Call;
import okhttp3.FormBody;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import okhttp3.ResponseBody;

import org.json.JSONObject;

import tw.y_studio.ptt.utils.DebugUtils;

import java.util.HashMap;
import java.util.Map;

public class SetPostRankAPIHelper extends BaseAPIHelper {

    public SetPostRankAPIHelper(Context context, String board, String aid) {
        super(context);
        this.board = board;
        this.aid = aid;
        this.data = new HashMap<>();
    }

    private Map<String, Object> data;
    private String board = "";
    private String aid = "";

    public Map<String, Object> getData() {
        return data;
    }

    public enum iRank {
        like,
        dislike,
        non,
    }

    private int iRank2Int(iRank rank) {
        switch (rank) {
            case like:
                return 1;
            case dislike:
                return -1;
            case non:
            default:
                return 0;
        }
    }

    public SetPostRankAPIHelper get(String pttid, iRank rank) throws Exception {
        data.clear();

        RequestBody body = new FormBody.Builder().build();
        Request request =
                new Request.Builder()
                        .post(body)
                        .url(
                                hostUrl
                                        + "/api/Rank/"
                                        + board
                                        + "/"
                                        + aid
                                        + "?pttid="
                                        + pttid
                                        + "&rank="
                                        + iRank2Int(rank))
                        .build();
        DebugUtils.Log("SetPostRankAPIHelper", "" + request.toString());
        Call mcall = getOkHttpClient().newCall(request);

        Response response = mcall.execute();
        final int code = response.code(); // can be any value
        if (!response.isSuccessful() && code != 200) {
            // error
            throw new Exception("Error Code : " + code);
        } else {
            ResponseBody mRb = response.body();
            String cont = mRb.string();
            JSONObject all = new JSONObject(cont);
            // Log.d("API","GetRankByPost = "+all.toString());
            data.put("Rank", all.getInt("rank"));
            data.put("PTTID", all.getString("pttid"));
            data.put("no", all.getString("no"));
            data.put("Board", all.getString("board"));
            data.put("AID", all.getString("aid"));
        }

        return this;
    }
}

File: app/src/main/java/tw/y_studio/ptt/api/SetPostRankAPIHelper.kt
package tw.y_studio.ptt.api

import android.content.Context
import okhttp3.FormBody
import okhttp3.Request
import okhttp3.RequestBody
import org.json.JSONObject
import tw.y_studio.ptt.utils.Log
import java.util.*
import kotlin.jvm.Throws

class SetPostRankAPIHelper() : BaseAPIHelper() {
    private val data: MutableMap<String, Any> = mutableMapOf()
    private val board = ""
    private val aid = ""
    fun getData(): Map<String, Any> {
        return data
    }

    enum class iRank {
        like, dislike, non
    }

    private fun iRank2Int(rank: iRank): Int {
        return when (rank) {
            iRank.like -> 1
            iRank.dislike -> -1
            iRank.non -> 0
            else -> 0
        }
    }

    @Throws(Exception::class)
    fun get(pttid: String, rank: iRank): SetPostRankAPIHelper {
        data.clear()
        val body: RequestBody = FormBody.Builder().build()
        val request = Request.Builder()
                .post(body)
                .url(
                        hostUrl
                                + "/api/Rank/"
                                + board
                                + "/"
                                + aid
                                + "?pttid="
                                + pttid
                                + "&rank="
                                + iRank2Int(rank))
                .build()
        Log("SetPostRankAPIHelper", "" + request.toString())
        val mcall = okHttpClient!!.newCall(request)
        val response = mcall.execute()
        val code = response.code // can be any value
        if (!response.isSuccessful && code != 200) {
            // error
            throw Exception("Error Code : $code")
        } else {
            val mRb = response.body
            val cont = mRb!!.string()
            val all = JSONObject(cont)
            // Log.d("API","GetRankByPost = "+all.toString());
            data["Rank"] = all.getInt("rank")
            data["PTTID"] = all.getString("pttid")
            data["no"] = all.getString("no")
            data["Board"] = all.getString("board")
            data["AID"] = all.getString("aid")
        }
        return this
    }
}